### PR TITLE
Ilya consumables add page

### PR DESCRIPTION
**Pull request by ilyaflaks**
# Description
consumables-add

Adds functionality for the user to navigate to /consumables/add, enter the required information (consumable name, description, measurement units, size) and create a new consumable type in the system.

The front end validates the input to have a name that is at least 4 characters long and a description that is at least 10 characters long. The back end checks if a consumable under the same name already exists.

## Related PRS (if any):
This frontend PR is related to the [back end PR#737](https://github.com/OneCommunityGlobal/HGNRest/pull/737)
To test this front end PR you need to checkout the #737 back end PR, branch **ilya_consumables_add_page**
…

## Main changes explained:
- /consumables/add route added to the front end router
- AddConsumables component includes a form with validation that makes a POST request to the backend
- The new component is styled to accommodate smaller screens 
- Necessary Redux actions and Reducers created
- Existing files formatted with ESLint and Prettier

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as any user
5. go to [http://localhost:3000/bmdashboard ](http://localhost:3000/bmdashboard) and log in again
6. go to [http://localhost:3000/bmdashboard/consumables/add](http://localhost:3000/bmdashboard/consumables/add), enter some dummy data, press "Add Consumable" when the button is enabled
7. A success message should appear if everything is acceptable or a warning message if not (such as a duplicate value)
consumables-fe
8. In MongoDB Compass open hgnData_dev.buildingInventoryTypes collection. Press "Find", scroll all the way to the bottom or search by name and verify that the entered information is saved in the database
compass

## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/a5600668-ccf6-4e3b-b5e7-b52be3e7cb7f

## Note:
Examples of construction consumables can include nails, screws, batteries, lumber, etc.

### CODE REVIEW

**Review by Pandani07 — DISMISSED** [review]
Tested this PR and i could add a new consumable. I got a screen success message after I added a new entry.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44672833/62e1f610-5885-4f35-b94d-779a7436a798

**Comment by meetpadhiar4** [comment]
I have tested this PR and all the changes and functionalities are working properly. Please see the attached video for reference.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/62424d0a-d0ec-437b-94fc-5de71f04f2e6

**Review by meetpadhiar4 — DISMISSED** [review]
I have tested this PR and all the changes and functionalities are working properly. Please see the attached video for reference.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/dda6cc1e-6225-4bd0-a7fc-afaf5fb2316b

**Review by ClemarNunes — DISMISSED** [review]
I tested your PR and everything worked as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/134466562/2e0b8748-e2ea-4d88-bda3-fc816f40fc74

**Review by ShengjieMao — DISMISSED** [review]
No problem with either the front/backend. Function worked as expected, and the check for inputs also considered the different cases. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56333697/c34ce231-35fb-43d9-8ffd-17f20ef49033

**Review by n0vay — DISMISSED** [review]
Looks Good to me
image

**Review by Renanluizssx — CHANGES_REQUESTED** [review]
Hi @ilyaflaks 
, i followed your steps but in my case, the message box appear an error, look my review please:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/8789db30-0f50-400b-9b0e-8e8d425ac0e2

**Review by Renanluizssx — DISMISSED** [review]
Hi @ilyaflaks, i made a mistake, now i added in correct branch, see my review:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/f7719a1e-abd0-4e54-aae3-f1d2f6ce0fed

**Review by SaiDeepak-NEU — CHANGES_REQUESTED** [review]
Hello Ilya,
I have tested the changes mentioned in the PR and found one issue.

When I am entering an empty object in the consumable name by just entering space 4 times, it is still considering it as an accepted value.
The same goes with description and size. The button is getting enabled and I am able to create a record for empty values . Please see the attached screen shot.
Screenshot (86)
Rest everything worked fine. successful popup on correct record creation and warning message on duplicate record creation.
Screenshot (85)
Screenshot (84)

**Comment by TareqMia** [comment]
Hi @ilyaflaks, I tested your PR with the associated backend PR and everything worked as expected. However, adding on to what Sai mentioned above, the Consumable Name, Description, and Size fields can all be submitted using the minimum number of required characters (ex. 4 for name). Other than that, great job on this 👍 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/39747003/de595939-43a7-4c27-8b44-347618179771

Screenshot 2024-02-10 at 7 56 40 PM

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/39747003/3031077b-ff03-4f94-b49e-8acd544c6dc6

**Review by 666saofeng — APPROVED** [review]
Hi, I tested with admin account and it worked as intended.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62363348/c9d4c590-aa29-4ea7-a63a-0daa07c641b4


image

**Comment by ilyaflaks** [comment]
> Hello Ilya, I have tested the changes mentioned in the PR and found one issue.
> 
> When I am entering an empty object in the consumable name by just entering space 4 times, it is still considering it as an accepted value. The same goes with description and size. The button is getting enabled and I am able to create a record for empty values . Please see the attached screen shot. Screenshot (86) Rest everything worked fine. successful popup on correct record creation and warning message on duplicate record creation. Screenshot (85) Screenshot (84)

@SaiDeepak-NEU and @TareqMia Thank you for catching that! I added non-white space validation on the front end to "name", "description" and "size" fields. Please review when you have a chance

**Comment by mianmiantea2019** [comment]
I have tested this PR and some edge cases, functionalities are working properly. Good work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/59427068/e47b37f6-dca1-452a-bf23-02cdf6372505

**Review by peterson337 — DISMISSED** [review]
Hi ilyaflaks, I tested your application, and everything worked as expected. Good job!

one comunnity

**Review by Jingyii800 — DISMISSED** [review]
Hi @ilyaflaks, I have tested this feature with the backend PR, and it works as expected. Great work! I only have one concern, when I enter digits and special characters into "consumable name", it could also be recognized as value. I am wondering if you designed to accept this. Please check more details from my video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112589476/600f0b4a-4873-41e0-b37c-e23a0de660e4

**Review by brandta-1 — DISMISSED** [review]
# Review
Tested functionality using MongoDB and front end web application. Successfully logged into the building management dashboard and created a consumable. When the consumable was created, the application logged ```! vali err```. However the document was successfully created in MongoDB. Referenced this review on the [back end](https://github.com/OneCommunityGlobal/HGNRest/pull/737#pullrequestreview-1887074584).

# Video

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/116298512/1de9ac3d-d6c3-4744-a3f4-054a3ba944f0


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/116298512/ec8d6e6a-05ae-482e-807e-d6302ac8a437

**Comment by ilyaflaks** [comment]
@Jingyii800 Thanks for bringing up the issue of entering digits and special characters in the name. I have discussed this with the team manager and we agreed that the validation needs to allow these characters. We can not predict all of the possible scenarios but some consumables might have those characters in the name. For example, WD-40 has both a number and a character. 

@brandta-1 Looks like I forgot to delete a console log! Thanks for catching that, I'll remove it shortly

**Review by XiaohanMeng — DISMISSED** [review]
Hi, I tested this PR and found that it functions excellently—well done! I verified the following:
* Successfully added a consumable and received a confirmation prompt upon submission.
* Attempted to add a consumable with the same name and received a duplication warning.
* Confirmed that only the successfully added consumables are visible in MongoDB Compass.
Please refer attached video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/0e91b37d-6598-40e5-8c5a-539e1d808e3a

**Review by 20chen-7 — DISMISSED** [review]
Everything looks good, and I could find the created inventory type in the MongoDB.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102696148/447596b9-615a-4183-9d4f-a577c27771ff

duplicated name
Screenshot 2024-03-17 at 12 31 37 AM
Screenshot 2024-03-17 at 12 48 23 AM

**Comment by ilyaflaks** [comment]
@one-community can you please merge this PR into "development"? All of the changes requested have been addressed and I resolved the merge conflicts. Let's close this before more merge conflicts arise. Thanks

File: src/actions/bmdashboard/invTypeActions.js
Comment (MQChen211) [inline src/actions/bmdashboard/invTypeActions.js:1]: Hi Ilya, I tested the PR and successfully added a new consumable and got a message. Here's the screenshot.

**Review by MQChen211 — APPROVED** [review]

**Pull request merged**